File: Linked List Cycle - Leetcode 141/Linked List Cycle - Leetcode 141.cpp
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode dummy;
        dummy.next = head;
        ListNode *slow = &dummy, *fast = &dummy;

        while (fast && fast->next) {
            slow = slow->next;
            fast = fast->next->next;

            if (slow == fast) {
                return true;
            }
        }

        return false;
    }
};

File: Linked List Cycle - Leetcode 141/Linked List Cycle - Leetcode 141.js
var hasCycle = function(head) {
    const dummy = new ListNode(0);
    dummy.next = head;
    let slow = dummy, fast = dummy;

    while (fast && fast.next) {
        slow = slow.next;
        fast = fast.next.next;

        if (slow === fast) {
            return true;
        }
    }

    return false;
};
